components/counter-bar/src/index.js

```diff
@@ -63,166 +61,172 @@ const Counter = styled('button')(
-  const totalScore = counters.reduce((a, b) => a + (Number(b.score) || 0), 0);
+const CounterBar = ({ children, ...props }) => {
   return (
-    <Container {...props}>
-      <CountWrapper active={activeTitle} disabled={!totalScore} onClick={() => onSelect()}>
-        <ResultCountTitle
-          count={totalScore}
-          countColor={totalScore > 0 ? WHITE : BLACK}
-          countBackgroundColor={totalScore > 0 ? RED : GREY_3}
-        >
-          {listTitle}
-        </ResultCountTitle>
-      </CountWrapper>
-      {counters.map(counter => (
-        <Counter
-          active={counter.active}
-          disabled={!counter.score}
-          key={counter.id}
-          name={counter.id}
-          onClick={() => onSelect(counter.id)}
-          empty={!counter.name || counter.name.length === 0}
-        >
-          <ResultCountTitle
-            countColor={counter.score > 0 ? WHITE : BLACK}
-            countBackgroundColor={counter.score > 0 ? RED : GREY_3}
-            count={counter.score || 0}
-          >
-            {counter.name}
-          </ResultCountTitle>
-        </Counter>
-      ))}
-    </Container>
+    <Wrapper {...props}>
+      {children}
+    </Wrapper>
   );
 };
 
 CounterBar.propTypes = {
-  activeTitle: PropTypes.bool,
-  counters: PropTypes.arrayOf(
-    PropTypes.shape({
-      active: PropTypes.bool,
-      id: PropTypes.any.isRequired,
-      name: PropTypes.string,
-      score: PropTypes.number,
-    }),
-  ),
-  listTitle: PropTypes.any,
-  name: PropTypes.string.isRequired,
+  children: PropTypes.node.isRequired,
+};
+
+CounterBar.Total = ({active, children, onSelect, score}) => {
+  return (
+    <TotalWrapper active={active} disabled={!score}>
+      <ResultCountTitle
+        count={score}
+        countColor={score > 0 ? WHITE : BLACK}
+        countBackgroundColor={score > 0 ? RED : GREY_3}
+        >
+        {children}
+      </ResultCountTitle>
+    </TotalWrapper>
+  );
+};
+CounterBar.Total.displayName = 'Total';
+
+CounterBar.Item = ({active, children, onSelect, score}) => {
+ return (
+   <ItemWrapper
+     active={active}
+     disabled={!score}
+     onClick={(e) => onSelect(e)}
+     empty={!children || children.length === 0}>
+     <ResultCountTitle
+       countColor={score > 0 ? WHITE : BLACK}
+       countBackgroundColor={score > 0 ? RED : GREY_3}
+       count={score || 0}
+          >
+       {children}
+     </ResultCountTitle>
+   </ItemWrapper>
+ ); 
+};
+CounterBar.Item.displayName = 'Item';
+
+const sharedPropTypes = {
+  active: PropTypes.bool,
+  children: PropTypes.node,
   onSelect: PropTypes.func,
+  score: PropTypes.number,
 };
 
-CounterBar.defaultProps = {
+const sharedDefaultProps = {
   onSelect: () => {},
 };
 
+CounterBar.Total.propTypes = sharedPropTypes;
+CounterBar.Total.defaultProps = sharedDefaultProps;
+CounterBar.Item.propTypes = sharedPropTypes;
+CounterBar.Item.defaultProps = sharedDefaultProps;
+
+CounterBar.Container = ({children}) => {
```

Comment:
this will throw away other props - preventing for example a `className` or `name` being set

should instead be `CounterBar.Container = props => <ItemsWrapper {...props} />;`

Reply:
or indeed just `CounterBar.Container = ItemsWrapper;`

Reply:
Great! I notice I can also do `const CounterBar = Wrapper`.

Reply:
...just as you mentioned above!